docs/architecture-description.md

```diff
@@ -719,20 +719,54 @@ l1RollingHashes(
 
 # Gas price setting
 
-The gas price for transaction execution in Linea is computed from L1 gas price.
-
-
-![gas price API](assets/gasPrice.drawio.svgo.svg)
-
-
-Gas price is based on the history of the gas price on L1.
-
-It uses the geth method `miner_setGasPrice` and the besu method
-
-`miner_setMinGasPrice` to update the gas price of the nodes.
-
-Both methods expect a single parameter: `0x${gasPrice.toString(16)}`
-
+There are 3 aspects of gas pricing on Linea:
+* Ensure Sequencer's inclusion logic is aligned to L1 fee market. This is to avoid exploiting Linea to execute
+transactions for unsustainably low fees
+* Ensure the best user experience for users who use Linea nodes with extended capabilities. Unlike vanilla Ethereum
+protocol, gas price on Linea and other rollups is not 2-dimensional (base fee, priority fee). There's at least L1 fees
+(execution fees and blob fees), infrastructural cost (mostly proving, but not only), potential priority fee
+(only when there's a high congestion and there's competition for L2 block space). This is an issue for interoperability,
+because vanilla Ethreum API isn't tailored for this. That's why there's a Besu plugin addressing this issue and
+providing gas price depending on input transaction
+* Ensure an OK user experience for users who run vanilla nodes. Namely, eth_gasPrice should return fees high enough, so
+99.9% of transactions are includable on Linea.
+
+This is how these challenges were solved technically:
+
+![gas price API](assets/gasPrice.drawio.svg)
+
+Coordinator fetches L1 fees data, based on which it will compute gas pricing components. There are 3:
+* Fixed cost. Represents infrastructural cost per unit of L2 gas. Doesn't really depend on L1 and it's just a
+configuration in Coordinator
+* Variable cost. Cost of a 1 byte of compressed data on L2, which is finalized on L1 contract. Depends on the fees Linea
+pays for finalization, which in turn depends on L1 blob and execution fee market
+* Legacy cost. Recommended gas price for vanilla Ethereum API (`eth_gasPrice`)
+
+## Gas pricing propagation
+This information is delivered to nodes 2 ways:
+* via extraData, part of vanilla Ethereum Protocol
+* via RPC calls (only Geth and Besu are supported and tested)
+
+### ExtraData
+Coordinator sends extraData to Sequencer via `miner_setExtraData`. ExtraData contains all 3 fields mentioned above.
+Sequencer in turn uses this information for inclusion logic, to include only profitable transactions and it adds last
+received extraData to the next block it seals. Once it gets inside a block, it's propagated to all the nodes on Linea
+via P2P as a block header's field. And since this info is on all the nodes, they can use this information to figure out,
+what's gas price for a given transaction that would make it includable on Linea. This currently is possible with Besu +
```

Comment:
`what's gas price for a given transaction that would make it includable on Linea` could be

`what the gas price is for a given transaction that would make it includable on Linea`